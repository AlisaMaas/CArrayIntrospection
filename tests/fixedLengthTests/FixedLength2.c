/**
* This tests that non-induction variables are ignored
* (for now - this test may be later used to make sure they aren't!)
**/
int foo(int* array){
	int sum = 0;
	for(int i = 0, j = 0;  40 > i && j < 50;  i++, j++){
		sum += array[i + 4];
		sum += array[j];
	}
	return sum;
	}
		