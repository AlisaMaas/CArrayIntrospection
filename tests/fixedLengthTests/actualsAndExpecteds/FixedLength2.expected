Printing analysis 'Promote Memory to Register' for function 'foo':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Canonicalize natural loops':
Pass::print not implemented for pass: 'Canonicalize natural loops'!
Printing analysis 'Induction Variable Simplification':
Pass::print not implemented for pass: 'Induction Variable Simplification'!
Printing analysis 'Loop Invariant Code Motion':
Pass::print not implemented for pass: 'Loop Invariant Code Motion'!
Printing analysis 'Simple constant propagation' for function 'foo':
Pass::print not implemented for pass: 'Simple constant propagation'!
Printing analysis 'Replace indexes using the induction variable with the constant upper bound of the loop, if it exists.':
Printing analysis 'Simple constant propagation' for function 'foo':
Pass::print not implemented for pass: 'Simple constant propagation'!
Printing analysis 'Find loops that have a fixed number of iterations that also index into arrays.':
Analyzing foo
Argument array has max index 43
